daemon/links/links_test.go

```diff
@@ -197,16 +197,16 @@ func TestLinkPortRangeEnv(t *testing.T) {
 		tcpport := fmt.Sprintf("DOCKER_PORT_%d_TCP+PORT", i)
 		tcpproto := fmt.Sprintf("DOCKER_PORT_%d_TCP+PROTO", i)
 		tcp := fmt.Sprintf("DOCKER_PORT_%d_TCP", i)
-		if env[tcpaddr] == "172.0.17.2" {
+		if env[tcpaddr] != "172.0.17.2" {
```

Comment:
So clearly something is completely wrong about this part of the test (and really difficult to know what the intent was).
The loop should be `for _, i := range …` and thus you're change should be OK. But I wonder if the intent of the test was to try out `DOCKER_PORT_1_TCP_ADDR` instead ?

PS : `tcpport` and `tcpproto` should have `_` instead of `+` here. I think it was a typo too